### ISSUE TITLE
Nodes created endless

### ISSUE THREAD

**Issue by ljhhuxiaoba** [memgraph/memgraph#1333]
I executed a query on my memgraph database and get a crash.

memgraph version:2.11.0
operating system: windows 11
API：Cypher version 9

To reproduce:
CREATE (),();
MERGE (n0) MERGE (n1) CREATE (c0);

Expected behavior:
Some new nodes should be created.

Actual behavior:
New nodes are created without stopping.

**Comment by katarinasupe** [comment]
Hi @ljhhuxiaoba, thank you for opening issues on the Memgraph repository and helping us improve our product. We value your input and would love to help out with your project. Would you be willing to hop on a quick call with me (the Developer Experience team) to speed up your work? My email is katarina.supe@memgraph.io if you need to get in touch and you can [join our Discord server](https://www.discord.gg/memgraph) for any additional questions 😄

**Comment by as51340** [comment]
Yep I agree, this is the issue connected with cypher semanti. I have a fix for this scenario but then few other edge cases failed so we will not tackle this issue now.
